test/e2e/declarative_test_fwk/helper.py

```diff
@@ -201,16 +235,50 @@ def assert_items(self, expectations: dict, state: dict) -> None:
         if not expectations:
             # nothing to assert as there are no expectations
             return
-        if not state:
-            # there are expectations but no given state to validate
+        if not actual:
+            # there are expectations but no given actual state to validate
             # following assert will fail and assert introspection will provide useful information for debugging
-            assert expectations == state
+            assert expectations == actual
 
         for (key, value) in expectations.items():
             # conditions are processed separately
             if key == "conditions":
                 continue
-            assert (key, value) == (key, state.get(key))
+            # tags are processed by custom logic
+            if key == "tags":
+                for expectedTag in value:
+                    tagKeyExist = False
+                    tagKey = expectedTag.get("key")
+                    latestTags = actual.get(key)
+                    for latestTag in latestTags:
+                        if ("key", tagKey) in latestTag.items():
+                            assert expectedTag == latestTag
+                            tagKeyExist = True
+                            break
+                    if tagKeyExist:
+                        continue
+                    assert expectedTag == None
+                continue
```

Comment:
How about the tag value?

Reply:
expectedTag and latestTag include both tag key and tag value